opencog/spacetime/octomap/OctoMapValue.cc

```diff
@@ -0,0 +1,58 @@
+#include "OctoMapValue.h"
+
+#include <opencog/spacetime/octomap/atom_types.h>
+#include <opencog/util/exceptions.h>
+#include <opencog/atoms/base/ClassServer.h>
+#include <opencog/atoms/value/ValueFactory.h>
+
+using namespace opencog;
+
+
+void OctoValue::update() const
+{
+    auto it = (_om->get_locations_of_atom(_item)).rbegin(); //We need the last.
+    _value = {it->x(), it->y(), it->z()};
+}
+
+OctoValue::OctoValue(const HandleSeq& hseq) : FloatValue(OCTO_VALUE)
+{
+
+    if( hseq.size() != 2) {
+        throw InvalidParamException(TRACE_INFO, 
+                "Required 2 arguments. Provided %d.", hseq.size());
+    }
+
+    _item = hseq[0];
+    _octo_node = hseq[1];
+    _om = OctoMapNodeCast(_octo_node)->get_map();
```

Comment:
I'm fairly certain you want to have `_om->insert_atom(_item);` here.  Maybe you are hoping that "someone else" will do the insert, but that is a poor thing to hope for, because "someone else" might forget.  In that case, you end up with the confusing situation where the octomap isn't actually tracking the node that you are given, here.

Reply:
never mind. I see that the octomap API doesn't allow that.